src/main/java/org/folio/innreach/batch/contribution/service/OngoingContributionEventProcessor.java

```diff
@@ -0,0 +1,94 @@
+  private final JsonHelper jsonHelper;
+  private final OngoingContributionStatusService ongoingContributionStatusService;
+  private final TenantScopedExecutionService executionService;
+  @Value("${contribution.retry-attempts}")
+  private int maxRetryAttempts;
+
+  @Async("ongoingSchedulerTaskExecutor")
+  public void processOngoingContribution(OngoingContributionStatus ongoingContributionStatus) {
+    try {
+      log.info("processOngoingContribution:: Processing ongoing contribution event with id {} , tenant {}",
+        ongoingContributionStatus.getId(), ongoingContributionStatus.getTenant());
+      executionService.executeAsyncTenantScoped(ongoingContributionStatus.getTenant(),
+        () -> {
+          checkRetryLimit(ongoingContributionStatus);
+          switch (ongoingContributionStatus.getDomainEventName()) {
+            case ITEM -> processItem(ongoingContributionStatus);
+            case HOLDINGS -> processHoldings(ongoingContributionStatus);
+            case INSTANCE -> processInstance(ongoingContributionStatus);
+            default ->
+              ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, UNKNOWN_EVENT_NAME_MESSAGE, FAILED);
+          }
+        });
+    } catch (ServiceSuspendedException | InnReachConnectionException |
+             SocketTimeOutExceptionWrapper | InnReachGatewayException ex) {
+      ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, RETRY);
+    } catch (Exception ex) {
+      ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, ex.getMessage(), FAILED);
+    }
+  }
+
+  private void processItem(OngoingContributionStatus ongoingContributionStatus) {
+    Item oldEntity = jsonHelper.fromJson(ongoingContributionStatus.getOldEntity(), Item.class);
+    Item newEntity = jsonHelper.fromJson(ongoingContributionStatus.getNewEntity(), Item.class);
+    log.info("processItem:: processing item with oldEntity {} and newEntity {}", oldEntity, newEntity);
+    switch (ongoingContributionStatus.getDomainEventType()) {
+      case CREATED -> contributionActionService.handleItemCreation(newEntity, ongoingContributionStatus);
+      case UPDATED -> {
+        contributionActionService.handleItemUpdate(newEntity, oldEntity, ongoingContributionStatus);
+        transactionActionService.handleItemUpdate(newEntity, oldEntity);
+      }
+      case DELETED -> contributionActionService.handleItemDelete(oldEntity, ongoingContributionStatus);
+      default ->
+        ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, UNKNOWN_TYPE_MESSAGE, FAILED);
+    }
+  }
+
+  private void processHoldings(OngoingContributionStatus ongoingContributionStatus) {
+    log.info("Not yet implemented {}", ongoingContributionStatus);
+  }
+
+  private void processInstance(OngoingContributionStatus ongoingContributionStatus) {
+    log.info("Not yet implemented {}", ongoingContributionStatus);
```

Comment:
I would add some prefix to both of these log messages to distinguish which is being called.

Reply:
Done